File: platforms/web/examples/core/pass/set_viewport.js

import { Renderer, Pass, Shader } from "fragmentcolor";

const renderer = new Renderer();
const target = await renderer.createTextureTarget([64, 64]);

const shader = Shader.default();
const pass = new Pass("clipped");
pass.addShader(shader);

pass.setViewport({ x: 0, y: 0, width: 32, height: 32 });

renderer.render(pass, target);

File: platforms/python/examples/core/pass/set_viewport.py

from fragmentcolor import Renderer, Pass, Shader, Region

renderer = Renderer()
target = renderer.create_texture_target([64, 64])

shader = Shader.default()
rpass = Pass("clipped")
rpass.add_shader(shader)

rpass.set_viewport(Region((0, 0), (32, 32)))

renderer.render(rpass, target)
